File: thesis/src/sources/echo/context_0_0.py
import values
import sys
value_file = open("Ft9AKVWPU_values.txt", "w")
exception_file = open("Ft9AKVWPU_exceptions.txt", "w")
def write_separator():
    value_file.write("--Ft9AKVWPU-- SEP")
    exception_file.write("--Ft9AKVWPU-- SEP")
    sys.stderr.write("--Ft9AKVWPU-- SEP")
    sys.stdout.write("--Ft9AKVWPU-- SEP")
    sys.stdout.flush()
    sys.stderr.flush()
    value_file.flush()
    exception_file.flush()
def send_value(value):
    values.send_value(value_file, value)
def send_exception(exception):
    values.send_exception(exception_file, exception)
def send_specific_value(value):
    values.send_evaluated(value_file, value)
def send_specific_exception(exception):
    values.send_evaluated(exception_file, exception)
try:
    write_separator()
    from submission import *
except Exception as e:
    send_exception(e)
else:
    send_exception(None)
value_file.close()
exception_file.close()

File: thesis/src/sources/echo/Context00.hs
{-# LANGUAGE NamedFieldPuns #-}
module Context00 where
import System.IO (hPutStr, stderr, stdout, hFlush)
import System.Environment
import qualified Values
import Control.Monad.Trans.Class
import Control.Exception
import EvaluationUtils
import qualified Submission
value_file = "EYo0SKDXN_values.txt"
exception_file = "EYo0SKDXN_exceptions.txt"
writeSeparator :: IO ()
writeSeparator = do
    hPutStr stderr "--EYo0SKDXN-- SEP"
    hPutStr stdout "--EYo0SKDXN-- SEP"
    appendFile value_file "--EYo0SKDXN-- SEP"
    appendFile exception_file "--EYo0SKDXN-- SEP"
    hFlush stdout
    hFlush stderr
sendValue :: Values.Typeable a => a -> IO ()
sendValue = Values.sendValue value_file
sendException :: Exception e => Maybe e -> IO ()
sendException = Values.sendException exception_file
sendSpecificValue :: EvaluationResult -> IO ()
sendSpecificValue = Values.sendEvaluated value_file
sendSpecificException :: EvaluationResult -> IO ()
sendSpecificException = Values.sendEvaluated exception_file
handleException :: Exception e => (Either e a) -> Maybe e
handleException (Left e) = Just e
handleException (Right _) = Nothing
main = do
    writeSeparator
    let mainArgs = {}
    result <- try (withArgs mainArgs Submission.main) :: IO (Either SomeException ())
    let ee = handleException result in sendException (ee)
    putStr ""